core/src/kura.rs

```diff
@@ -991,4 +1039,26 @@ mod tests {
             .init()
             .unwrap();
     }
+
+    #[test]
+    #[should_panic]
+    fn write_then_read_wsv_from_disk_should_fail() {
+        let dir = tempfile::tempdir().unwrap();
+        let kura = Kura::blank_kura_for_testing_with_dir(&dir.path());
+        let mut wsv = WorldStateView::new(crate::World::default(), kura.clone());
+
+        std::thread::sleep(std::time::Duration::from_secs(1));
+        let loaded_data = kura.try_load_wsv().unwrap();
+    }
+
+    #[test]
+    fn write_then_read_wsv_from_disk() {
+        let dir = tempfile::tempdir().unwrap();
+        let kura = Kura::blank_kura_for_testing_with_dir(&dir.path());
+        let mut wsv = WorldStateView::new(crate::World::default(), kura.clone());
+
+        kura.store_wsv(&wsv);
+        std::thread::sleep(std::time::Duration::from_secs(1));
```

Comment:
Oof, kind of unstable. Consider refactor so that user can decide if they want to detach or join thread